sdk/typescript/bcs/package.json

```diff
@@ -4,6 +4,7 @@
   "description": "BCS - Canonical Binary Serialization implementation for JavaScript",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
+  "module": "dist/bcs.esm.js",
```

Comment:
Why does this PR include changes to the TypeScript SDK?

Reply:
It wasn't exporting the correct ESM module, which made it fail when being bundled with Vite. This will get broken out into a separate PR though (likely #4143)